### ISSUE TITLE
DocumentationStyleBear should display a list of supported languages

### ISSUE THREAD

**Issue by kriti21** [coala/coala-bears#2273]
DocumentationStyleBear prompts the user for language to use on the file. User can enter multiple options for the same thing like Python, python2 or python3 or C++ instead of cpp etc. Better would be if the list of supported languages are displayed and the user can choose and type the one applicable.

**Comment by Makman2** [comment]
> Better would be if the list of supported languages are displayed and the user can choose and type the one applicable.

So coala runs and the user is queried when he has duplicated languages?
I would rather
- throw an error quickly, or
- Raise a warning and ignore that (both language strings refer to the same language)

Another query would make coala even more cluttered, and as this is settings-related, it would be unusual to have a special exception for the language setting (there are probably other settings for bears which expose similar behaviour).

**Comment by kriti21** [comment]
DocumentationStyleBear being generic asks the user to enter the language for which this bear is being used. So the problem is cpp is valid but C++ is not (bear does not support C++ and returns error). Similarly python is valid but python2 and python3 are not. Using a list of supported languages to select from would therefore make more sense. 
This requires only modification of the existing query. @Makman2

**Comment by Makman2** [comment]
The query is performed through the standard setting-collection facility, changing this is highly non-trivial and I would refrain from it for just this special case. The current query is a generic one that happens for any query where values are missing.

We could maybe add some suggestion support in general for setting-queries, but this is also not so easy and needs to be thought about intensively (how far shall this go? how many values can be presented so a user is not distracted? ...)

**Comment by jayvdb** [comment]
I agree the bear settings system needs to support `language` better, and especially needs to provide a way for bears to advertise the restricted list of languages it supports.

We have a similar problem with CPDBear, and almost any other bear which supports multiple languages will also have a similar problem, as there are usually some languages which a bear can not support as it would be illogical.

Perhaps aspects can help.

In CPDBear, there is a custom warning.  But the real solution needs to be in core.
